### ISSUE TITLE
 Quarkus Native on(GraalVM) issue with ElasticSearch Java Client: No deserializer found in 'co.elastic.clients.elasticsearch.core.UpdateByQueryRequest._DESERIALIZER'

### ISSUE THREAD

**Issue by AbhinabKanrar** [quarkusio/quarkus#33072]
### Describe the bug

Using `Quarkus 3.0.1.Final` with `ElasticSearch 8.7`.  I've written a small piece of code to use ES's `update_by_query` feature:
```
try (Reader query = new StringReader(/*my json query String*/)) {
  var request = UpdateByQueryRequest.of(fn -> fn.index(index).withJson(query));

  return client.updateByQuery(request);
} catch (IOException e) {
  throw new RuntimeException(e);
}
```
This actually works fine in normal jvm mode. However whenever I'm trying to run this is Quarkus Native(GraalVM) mode I'm getting Deserialization error.

Using Maven as the build tool and this is my dependencies:
```
	<dependencies>
		<dependency>
			<groupId>io.quarkus</groupId>
			<artifactId>quarkus-arc</artifactId>
		</dependency>
		<dependency>
			<groupId>io.quarkus</groupId>
			<artifactId>quarkus-resteasy-reactive</artifactId>
		</dependency>
		<dependency>
			<groupId>io.quarkus</groupId>
			<artifactId>quarkus-resteasy-reactive-jackson</artifactId>
		</dependency>
		<dependency>
			<groupId>io.quarkus</groupId>
			<artifactId>quarkus-elasticsearch-java-client</artifactId>
		</dependency>
	</dependencies>
```

### Expected behavior

In Quarkus native mode the ElasticSearch java client should work just like JVM mode

### Actual behavior

Getting this error:
```
ERROR [com.mab.cha.con.exc.ServerExceptionHandler] (vert.x-eventloop-thread-1) Unhandled error: java.lang.RuntimeException: No deserializer found in 'co.elastic.clients.elasticsearch.core.UpdateByQueryRequest._DESERIALIZER'
	at co.elastic.clients.json.JsonpMapperBase.findDeserializer(JsonpMapperBase.java:92)
	at co.elastic.clients.json.JsonpMapperBase.findDeserializer(JsonpMapperBase.java:78)
	at co.elastic.clients.util.WithJsonObjectBuilderBase.withJson(WithJsonObjectBuilderBase.java:44)
	at co.elastic.clients.json.WithJson.withJson(WithJson.java:57)
	at com.mabsisa.chalona.client.EsClient.lambda$0(EsClient.java:66)
	at co.elastic.clients.elasticsearch.core.UpdateByQueryRequest.of(UpdateByQueryRequest.java:212)
	at com.mabsisa.chalona.client.EsClient.updateByQuery(EsClient.java:66)
	at com.mabsisa.chalona.client.EsClient_Subclass.updateByQuery$$superforward(Unknown Source)
	at com.mabsisa.chalona.client.EsClient_Subclass$$function$$1.apply(Unknown Source)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:74)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:63)
	at io.smallrye.faulttolerance.FaultToleranceInterceptor.lambda$syncFlow$3(FaultToleranceInterceptor.java:257)
	at io.smallrye.faulttolerance.core.InvocationContext.call(InvocationContext.java:20)
	at io.smallrye.faulttolerance.core.Invocation.apply(Invocation.java:29)
	at io.smallrye.faulttolerance.core.retry.Retry.doApply(Retry.java:88)
	at io.smallrye.faulttolerance.core.retry.Retry.apply(Retry.java:42)
	at io.smallrye.faulttolerance.FaultToleranceInterceptor.syncFlow(FaultToleranceInterceptor.java:259)
	at io.smallrye.faulttolerance.FaultToleranceInterceptor.intercept(FaultToleranceInterceptor.java:186)
	at io.smallrye.faulttolerance.FaultToleranceInterceptor_Bean.intercept(Unknown Source)
	at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:42)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:38)
	at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:26)
	at com.mabsisa.chalona.client.EsClient_Subclass.updateByQuery(Unknown Source)
	at com.mabsisa.chalona.client.EsClient.updateByQuery(EsClient.java:77)
	at com.mabsisa.chalona.service.ProfileService.updateAboutMe(ProfileService.java:43)
	at com.mabsisa.chalona.routes.ProfileRoute.sync(ProfileRoute.java:43)
	at com.mabsisa.chalona.routes.ProfileRoute_Subclass.sync$$superforward(Unknown Source)
	at com.mabsisa.chalona.routes.ProfileRoute_Subclass$$function$$1.apply(Unknown Source)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:74)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:63)
	at io.quarkus.hibernate.validator.runtime.interceptor.AbstractMethodValidationInterceptor.validateMethodInvocation(AbstractMethodValidationInterceptor.java:71)
	at io.quarkus.hibernate.validator.runtime.jaxrs.ResteasyReactiveEndPointValidationInterceptor.validateMethodInvocation(ResteasyReactiveEndPointValidationInterceptor.java:21)
	at io.quarkus.hibernate.validator.runtime.jaxrs.ResteasyReactiveEndPointValidationInterceptor_Bean.intercept(Unknown Source)
	at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:42)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:38)
	at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:26)
	at com.mabsisa.chalona.routes.ProfileRoute_Subclass.sync(Unknown Source)
	at com.mabsisa.chalona.routes.ProfileRoute_ClientProxy.sync(Unknown Source)
	at com.mabsisa.chalona.routes.ProfileRoute$quarkusrestinvoker$sync_793ca314ec66d88518392f861de2144da4c4cb8f.invoke(Unknown Source)
	at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
	at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:141)
	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:145)
	at org.jboss.resteasy.reactive.server.vertx.VertxResteasyReactiveRequestContext$1$1.handle(VertxResteasyReactiveRequestContext.java:78)
	at org.jboss.resteasy.reactive.server.vertx.VertxResteasyReactiveRequestContext$1$1.handle(VertxResteasyReactiveRequestContext.java:75)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:264)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:246)
	at io.vertx.core.impl.EventLoopContext.lambda$runOnContext$0(EventLoopContext.java:43)
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:566)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base@17.0.7/java.lang.Thread.run(Thread.java:833)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:775)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.posix.thread.PosixPlatformThreads.pthreadStartRoutine(PosixPlatformThreads.java:203)
```

### How to Reproduce?

1. Use quarkus.platform.version `3.0.1.Final`
2. Add  `quarkus-elasticsearch-java-client` extension
3. Inject the ElasticsearchAsyncClient bean
```
@Inject
ElasticsearchAsyncClient client;
```
4. Call any search operation using this client with the `withJson` method. For example:
```
try (Reader query = new StringReader(/*my json query String*/)) {
  var request = UpdateByQueryRequest.of(fn -> fn.index(index).withJson(query));

  return client.updateByQuery(request);
} catch (IOException e) {
  throw new RuntimeException(e);
}
```
5. This should raise this exception: `ERROR [com.mab.cha.con.exc.ServerExceptionHandler] (vert.x-eventloop-thread-1) Unhandled error: java.lang.RuntimeException: No deserializer found in 'co.elastic.clients.elasticsearch.core.UpdateByQueryRequest._DESERIALIZER'`

### Output of `uname -a` or `ver`

quay.io/quarkus/ubi-quarkus-mandrel-builder-image:22.3-java17

### Output of `java -version`

openjdk 17.0.6 2023-01-17

### Quarkus version or git rev

3.0.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.8

**Comment by AbhinabKanrar** [comment]
Update time!!!! 

same query works fine when I'm using the same with the java classes like:
```
    var script =
        Script.of(fn -> fn.stored(StoredScriptId.of(ssi -> ssi.id(scriptId).params(params))));

    var request = UpdateByQueryRequest.of(fn -> fn.index(index).script(script).maxDocs(maxDocs)
        .conflicts(conflicts).refresh(refresh).ignoreUnavailable(ignoreUnavailable));

    return client.updateByQuery(request);
```
Hence, seem like this issue is happening only when using the famous `withJson` method

**Comment by yrodiere** [comment]
Yes, this is caused by `withJson`.

Apparently when you call that method, the [Elasticsearch Java client relies on reflection to dynamically retrieve the corresponding deserializer](https://github.com/elastic/elasticsearch-java/blob/main/java-client/src/main/java/co/elastic/clients/json/JsonpMapperBase.java#L84-L95) stored in a static variable of the class (generally named `_DESERIALIZER`, e.g. https://github.com/elastic/elasticsearch-java/blob/main/java-client/src/main/java/co/elastic/clients/elasticsearch/core/UpdateByQueryRequest.java#L1216-L1217 ).

These reflection calls are not detected by GraalVM, which assumes `_DESERIALIZER` is not used at all, and removes it from the resulting native image, leading to the runtime exception you described.

Generally we would solve that by registering the relevant classes for reflection, which would lead to GraalVM not removing those `_DESERIALIZER` fields from the native image.

But... as I started implementing something like that for classes annotated with `@co.elastic.clients.json.JsonpDeserializable`, I noticed this would basically mean registering almost all classes in the Elasticsearch Java Client (thousands of them). This will probably lead to massively larger native images, *even if you don't use `withJson` at all*.

I'm currently looking for a better way to handle this, one that relies on GraalVM's `Feature`, but I'm not entirely sure this will work. Will keep you updated.

Alternatively, these are the only other ways forward I can reasonably consider:

* Not supporting `withJson()` at all: documenting the limitation and perhaps making sure we throw an exception in native builds that try to use it.
* Reporting the problem upstream so that the Elasticsearch team considers an alternative implementation that does not rely on reflection and works better with GraalVM. That would be theoretically possible since I believe the client code is generated to some extent, but I'm not sure how easy that would be nor how much effort the Elasticsearch teams is willing to put into GraalVM support.
* Somehow detecting when a feature (such as `withJson()`) is reachable and only then enabling reflection for basically every class of the Java client. Then users of `withJson()` will pay a high price, but other users (most of them) won't have to.
